From 85b07b6325d3629b18caaa3530f165170ea1f928 Mon Sep 17 00:00:00 2001
From: Ewan Milne <emilne@redhat.com>
Date: Wed, 20 Aug 2014 18:58:51 -0400
Subject: [scsi] scsi_transport_fc: Add 32Gbps speed definition

Message-id: <1408561176-18133-13-git-send-email-emilne@redhat.com>
Patchwork-id: 88438
O-Subject: [RHEL7.1 PATCH e-stor 12/57] [SCSI] scsi_transport_fc: Add 32Gbps speed definition.
Bugzilla: 1132092
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Ewan D. Milne" <emilne@redhat.com>

upstream commit 624f28be81097d06796e12c4c10aa7611305b334

Tested by Ewan D. Milne and Barry Donahue with a combination
of manual and automated testing.  No issues seen.

upstream commit 624f28be81097d06796e12c4c10aa7611305b334
Author: Chad Dupuis <chad.dupuis@qlogic.com>
Date:   Wed Feb 26 04:15:05 2014 -0500

    [SCSI] scsi_transport_fc: Add 32Gbps speed definition.

    Signed-off-by: Chad Dupuis <chad.dupuis@qlogic.com>
    Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
    Signed-off-by: James Bottomley <JBottomley@Parallels.com>

Signed-off-by: Ewan D. Milne <emilne@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_transport_fc.c b/drivers/scsi/scsi_transport_fc.c
index e106c27..fe6aa8e 100644
--- a/drivers/scsi/scsi_transport_fc.c
+++ b/drivers/scsi/scsi_transport_fc.c
@@ -261,6 +261,7 @@ static const struct {
  { FC_PORTSPEED_10GBIT,  "10 Gbit" },
  { FC_PORTSPEED_8GBIT,  "8 Gbit" },
  { FC_PORTSPEED_16GBIT,  "16 Gbit" },
+ { FC_PORTSPEED_32GBIT,  "32 Gbit" },
  { FC_PORTSPEED_NOT_NEGOTIATED, "Not Negotiated" },
 };
 fc_bitfield_name_search(port_speed, fc_port_speed_names)
diff --git a/include/scsi/scsi_transport_fc.h b/include/scsi/scsi_transport_fc.h
index b797e8f..8c79980 100644
--- a/include/scsi/scsi_transport_fc.h
+++ b/include/scsi/scsi_transport_fc.h
@@ -130,6 +130,7 @@ enum fc_vport_state {
 #define FC_PORTSPEED_4GBIT  8
 #define FC_PORTSPEED_8GBIT  0x10
 #define FC_PORTSPEED_16GBIT  0x20
+#define FC_PORTSPEED_32GBIT  0x40
 #define FC_PORTSPEED_NOT_NEGOTIATED (1 << 15) /* Speed not established */
 
 /*
-- 
1.7.1